File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        ListNode list1 = headA;
        ListNode list2 = headB;
        
        int size1=0;
        int size2 =0;
        while(list1 !=null){
            list1 = list1.next;
            size1++;
        }
        while(list2 !=null){
            list2 = list2.next;
            size2++;
        }
        ListNode listA = headA;
        ListNode listB = headB;
        if(size1>size2){
            while(size1 != size2){
                listA = listA.next;
                size1--;
            }
        }else{
            while(size2!=size1){
                listB = listB.next;
                size2--;
            }
        }

        for(int i=0;i<size1;i++){
            if(listA == listB){
                return listA;
            }
            listA = listA.next;
            listB = listB.next;
        }
        return null;

    }
}   

File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        ListNode* temp1 = headA;
        ListNode* temp2 = headB;
        ListNode* vir1 = headA;
        ListNode* vir2 = headB;
        int size1 = 0, size2 = 0;
        while(temp1!=NULL){
            temp1 = temp1->next;
            size1++;
        }

        while(temp2!=NULL){
            temp2 = temp2->next;
            size2++;
        }
        
        int demo = 0;   

        if(size1>size2){
            demo = size1-size2;

            while(demo!=0){
                demo--;
                vir1 = vir1->next;
            }
        }
        else{
            demo = size2 - size1;
            while(demo!=0){
                demo--;
                vir2 = vir2->next;
            }
        }


       while (vir1 && vir2) {
            if (vir1 == vir2) return vir1;
            vir1 = vir1->next;
            vir2 = vir2->next;
        }

        return NULL;


        



    }
};